lib/jellyfish/room.ex

```diff
@@ -365,6 +366,31 @@ defmodule Jellyfish.Room do
     {:noreply, state}
   end
 
+  @impl true
+  def handle_info(%Membrane.RTC.Engine.Message.EndpointMessage{} = msg, state) do
+    Logger.debug("Received msg from endpoint: #{inspect(msg)}")
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info(
+        %EndpointRemoved{endpoint_id: endpoint_id},
+        state
+      )
+      when not is_map_key(state.components, endpoint_id) or
+             not is_map_key(state.peers, endpoint_id) do
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info(
+        %EndpointAdded{endpoint_id: endpoint_id},
+        state
+      )
+      when is_map_key(state.components, endpoint_id) or is_map_key(state.peers, endpoint_id) do
```

Comment:
Isn't this RC? When we add and remove an endpoint fast enough we can end up removing endpoint faster than reading a message from the engine? I am not sure but I would remove those guards and create one handle_info for all notifications (if possible)

Reply:
We always add/remove peers/components before receiving a notification about it, so there is no RC. This is also why we had logs `Unexpected info RoomAdded/RoomDeleted`. It would be hard to have one `handle_info` for all notifications as for example we handle `Message.EndpointMessage` with `media_events` separately.